[18.0][MIG] mis_builder_cash_flow: Migration to 18.0

geraldo29 opened the pull request:
Migration from 17.0 to 18.0

Issue: https://github.com/OCA/account-financial-reporting/issues/1230

Followed technical method explained in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-18.0.

Changes:
1)Removed migration folder
2)Changed 17 to 18 version in a few places
3)Changed tree to list in a few places.
4)Added test-requirements.txt

jjscarafia committed:
[12.0][ADD] mis_builder_cash_flow module

etobella committed:
[IMP] mis_builder_cash_flow: Add tests

jjscarafia committed:
[FIX] cash_flow: forecast accumulate next columns

For mis_builder to compute correctly bale we need accounts with user type include_initial_balance. We use receivable type which is the more accurate for this purpose

alvarorib committed:
Added translation using Weblate (Portuguese)

CarlesAntoli committed:
Added translation using Weblate (Catalan)

CarlesAntoli committed:
Added translation using Weblate (Spanish)

amcor committed:
[IMP] include partner_id in mis builder cash flow

amcor committed:
[IMP] partner_id in forecastline

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (52 of 52 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-mis_builder_cash_flow/pt/

MiquelRForgeFlow committed:
[IMP] mis_builder_cash_flow: black, isort, prettier

MiquelRForgeFlow committed:
[MIG] mis_builder_cash_flow: Migration to 13.0

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (53 of 53 strings)

Translation: account-financial-reporting-13.0/account-financial-reporting-13.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-13-0/account-financial-reporting-13-0-mis_builder_cash_flow/es_AR/

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 37.7% (20 of 53 strings)

Translation: account-financial-reporting-13.0/account-financial-reporting-13.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-13-0/account-financial-reporting-13-0-mis_builder_cash_flow/nl/

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 81.1% (43 of 53 strings)

Translation: account-financial-reporting-13.0/account-financial-reporting-13.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-13-0/account-financial-reporting-13-0-mis_builder_cash_flow/es/

pedrobaeza committed:
[FIX] mis_builder_cash_flow: Proper SQL expression + default for v13

- On v13, invoices are journal entries, so we have to explicitly exclude
  cancelled invoices from the expression. We leave draft ones, as they
  are good for forecasting.
- Default value for the company of the forecast line doesn't follow new
  specification, putting an incorrect value.
- Extra improvement: avoid a non needed casting of the fixed column type.

pedrobaeza committed:
[IMP] mis_builder_cash_flow: Consider states according selection

"Target Moves" option in MIS report instance was previously ignored.

Now it's taken into account, but still excluding cancelled entries.

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (55 of 55 strings)

Translation: account-financial-reporting-13.0/account-financial-reporting-13.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-13-0/account-financial-reporting-13-0-mis_builder_cash_flow/es_AR/

rvalyi committed:
[IMP] mis_builder_cash_flow: black, isort, prettier

rvalyi committed:
[MIG] mis_builder_cash_flow: Migration to 14.0

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (55 of 55 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-mis_builder_cash_flow/it/

JasminSForgeFlow committed:
[MIG] mis_builder_cash_flow: Migration to 15.0

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (55 of 55 strings)

Translation: account-financial-reporting-15.0/account-financial-reporting-15.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-15-0/account-financial-reporting-15-0-mis_builder_cash_flow/es/

aktiv-heli-kantawala committed:
[MIG] mis_builder_cash_flow: Migration to 16.0
Co-Authored-By: Stefan Ungureanu <stefan.ungureanu@tecnativa.com>

Commit:
[UPD] Update mis_builder_cash_flow.pot

OCA-git-bot committed:
oca-github-bot post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-mis_builder_cash_flow/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-mis_builder_cash_flow/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (54 of 54 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-mis_builder_cash_flow/es/

antoniospneto committed:
Added translation using Weblate (Portuguese (Brazil))

antoniospneto committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (54 of 54 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-mis_builder_cash_flow/pt_BR/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-mis_builder_cash_flow/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.1% (53 of 54 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-mis_builder_cash_flow/it/

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (54 of 54 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-mis_builder_cash_flow/pt_BR/

jakobkrabbe committed:
Added translation using Weblate (Swedish)

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 74.0% (40 of 54 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-mis_builder_cash_flow/sv/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-mis_builder_cash_flow/it/

pedrobaeza committed:
[OU-ADD] mis_builder_cash_flow: Migration scripts

TT49378

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (54 of 54 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-mis_builder_cash_flow/sv/

cormaza committed:
[IMP] mis_builder_cash_flow: pre-commit auto fixes

cormaza committed:
[17.0][MIG] mis_builder_cash_flow

Commit:
[UPD] Update mis_builder_cash_flow.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-mis_builder_cash_flow/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (53 of 53 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-mis_builder_cash_flow
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-mis_builder_cash_flow/es_AR/

victoralmau committed:
[OU-FIX] mis_builder_cash_flow: use env in openupgrade load_data instead of cr

As stated in https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L297, the param should `env`, instead of the cursor

Commit:
[MIG] mis_builder_cash_flow: Migration to 18.0

- Standard Odoo17->18 migration procedures
- Replace company_id with company_ids (Many2one → Many2many) in forecast lines
- Update constraint validation to check company overlap instead of exact match
- Modify SQL query to handle Many2many relationship with proper JOIN
- Update UI views to use many2many_tags widget for better UX
- Fix test_company_constrain to properly test new validation logic
- Update translations for new validation error message
- Apply code formatting fixes for linting compliance
- Fixes compatibility with Odoo 18's shared account model where
account.account now uses company_ids instead of company_id

geraldo29 commented:
Error: + pip install -r test-requirements.txt -c test-constraints.txt
Looking in indexes: https://wheelhouse.odoo-community.org/oca-simple-and-pypi
Collecting odoo-addon-date_range==18.0.* (from -r test-requirements.txt (line 1))
  Downloading https://wheelhouse.odoo-community.org/oca-simple/odoo-addon-date-range/odoo_addon_date_range-18.0.1.0.0.5-py3-none-any.whl (2.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 91.6 MB/s eta 0:00:00
ERROR: Could not find a version that satisfies the requirement odoo-addon-mis_builder==18.0.* (from versions: 15.0.1.0.1, 15.0.4.0.1, 15.0.4.0.2, 15.0.4.0.2.1, 15.0.4.0.3, 15.0.4.0.3.1, 15.0.4.0.3.2, 15.0.4.0.4, 15.0.4.0.5, 15.0.4.1.0, 15.0.4.1.0.1, 15.0.4.1.1, 15.0.4.1.2, 15.0.4.1.3, 15.0.4.1.3.1, 15.0.4.1.3.2, 15.0.4.1.3.3, 15.0.4.1.3.4, 15.0.4.1.3.5, 15.0.4.1.3.6, 15.0.4.1.3.7, 15.0.4.1.3.8, 15.0.4.1.4, 15.0.4.1.4.1, 15.0.4.1.4.2, 15.0.4.1.4.4, 15.0.4.1.5, 15.0.4.1.5.1, 15.0.4.1.5.2, 15.0.4.1.5.3, 15.0.4.1.5.4, 15.0.4.1.5.5, 15.0.4.1.5.6, 15.0.4.1.5.7, 15.0.4.1.5.8, 15.0.4.1.6, 15.0.4.1.6.1, 16.0.5.0.1, 16.0.5.0.1.1, 16.0.5.1.0.3, 16.0.5.1.0.4, 16.0.5.1.1, 16.0.5.1.1.1, 16.0.5.1.1.2, 16.0.5.1.1.4, 16.0.5.1.1.5, 16.0.5.1.2, 16.0.5.1.3, 16.0.5.1.3.2, 16.0.5.1.3.5, 16.0.5.1.3.7, 16.0.5.1.4, 16.0.5.1.4.3, 16.0.5.1.5, 16.0.5.1.5.1, 16.0.5.1.5.3, 16.0.5.1.6, 16.0.5.1.7, 16.0.5.1.7.2, 16.0.5.1.8, 16.0.5.1.9, 16.0.5.1.9.1, 16.0.5.1.9.2, 16.0.5.1.9.3, 16.0.5.1.9.4, 16.0.5.1.9.5, 16.0.5.1.9.6, 16.0.5.1.9.7, 16.0.5.1.9.8, 16.0.5.1.9.9, 16.0.5.1.10, 16.0.5.1.10.1, 16.0.5.1.11, 16.0.5.1.11.1, 16.0.5.1.11.2, 16.0.5.1.11.3, 16.0.5.1.11.6, 16.0.5.1.12, 16.0.5.1.12.1, 16.0.5.1.13, 16.0.5.2.0, 16.0.5.2.0.1, 16.0.5.2.1, 16.0.5.2.1.1, 16.0.5.2.2, 16.0.5.2.2.2, 16.0.5.2.2.3, 16.0.5.2.3, 16.0.5.2.3.1, 17.0.1.0.0.2, 17.0.1.0.0.3, 17.0.1.0.0.4, 17.0.1.0.0.6, 17.0.1.0.0.7, 17.0.1.0.1, 17.0.1.0.2, 17.0.1.0.3, 17.0.1.0.4, 17.0.1.0.4.1, 17.0.1.0.5, 17.0.1.1.0, 17.0.1.2.0, 17.0.1.2.0.1, 17.0.1.2.0.2, 17.0.1.2.1, 17.0.1.2.1.1)
ERROR: No matching distribution found for odoo-addon-mis_builder==18.0.*



No version 18 for https://wheelhouse.odoo-community.org/oca-simple/odoo-addon-mis-builder/

geraldo29 commented:
Previous error of no version 18 was already fixed and that worked. 

Changed company_id to company_ids manytomany

pedrobaeza commented:
Please squash all your migration commits into one.

/ocabot migration mis_builder_cash_flow

geraldo29 commented:
/ocabot migration mis_builder_cash_flow

geraldo29 commented:
Squashed commits. @pedrobaeza

geraldo29 commented:
Tested in my Odoo 18 instance and working as expected. Let me know if any changes are needed. Thanks
![WhatsApp Image 2025-06-15 at 23 06 18](https://github.com/user-attachments/assets/59b45bcd-898f-454b-91f5-0b6587e0079b)

pedrobaeza commented:
OK, thanks, but the commit message is not correct:

![imagen](https://github.com/user-attachments/assets/0fbb4ba8-23e5-40eb-894d-5fbfe8f401a2)

It should be changed to:

```
[MIG] mis_builder_cash_flow: Migration to 18.0

Extra explanation about the changes done if you want.
```

geraldo29 commented:
Changed commit message. Added the following: 

<img width="608" alt="image" src="https://github.com/user-attachments/assets/eb6ce744-e3f1-45e7-9e21-7ef9ebf89eb1" />

pedrobaeza commented:
Thanks for the changes. Testing the module in runboat, for forecast lines, the change of company_id to company_ids is not correct: the forecast line still should be to one company. And the field has been put as readonly. What you should do is to allow to search for accounts of that company (shared or not).

geraldo29 commented:
Fixed forecast line to be one company and editable. Allowed for search for accounts of that company shared or not. 

![image](https://github.com/user-attachments/assets/831c4024-9eac-4000-941a-c59543c7d172)
![image](https://github.com/user-attachments/assets/dfa99eef-91d5-4d74-b10e-85344b375080)

Squashed commits again.

pedrobaeza commented on mis_builder_cash_flow/report/mis_cash_flow.py:
```diff
@@ -84,7 +84,7 @@ def init(self):
             UNION ALL
             SELECT
-                fl.id as id,
+                fl.id + 1000000 as id,
```
Why do you need this? If there's a conflict of ids, you are only delaying 100k records the conflict, not solving it.

pedrobaeza commented:
@geraldo29 are you going to continue this migration? Please answer remaining questions.

luisDIXMIT commented on mis_builder_cash_flow/models/mis_cash_flow_forecast_line.py:
```diff
@@ -0,0 +1,51 @@
+
+    @api.constrains("company_id", "account_id")
+    def _check_company_id_account_id(self):
+        for line in self:
```
```suggestion
        if self.filtered(lambda x: x.company_id not in x.account_id.company_ids):
            raise ValidationError(
                _("The Company and the Company of the Account must be the same.")
            )
```
Why do you need loop all the forecast lines? If company _id field has changed to a company_ids field treat it as a list.

etobella replied:
If we check the implementation of filtered, the speed should be similar. So it has no sense to do the change IMO

luisDIXMIT commented on mis_builder_cash_flow/views/mis_cash_flow_forecast_line_views.xml:
```diff
@@ -0,0 +1,92 @@
+                        <field name="partner_id" />
+                        <field
+                            name="account_id"
+                            domain="[('company_ids', 'in', company_id)]"
```
I have seen that this becomes from version 17.0 but why is different the domain for account_id field between tree and form view? It has not sense.

luisDIXMIT commented on mis_builder_cash_flow/report/mis_cash_flow.py:
```diff
@@ -0,0 +1,124 @@
+            WHERE aml.parent_state != 'cancel'
+            UNION ALL
+            SELECT
+                fl.id + 1000000 as id,
```
Maybe a better approach, according to what Pedro suggested.
```suggestion
                fl.id * 2 + 1 as id,
```

luisDIXMIT replied:
I couldn't try this query in a correct way @etobella . How can i run it?

luisDIXMIT commented on mis_builder_cash_flow/report/mis_cash_flow.py:
```diff
@@ -0,0 +1,124 @@
+            SELECT
+                -- we use negative id to avoid duplicates and we don't use
+                -- ROW_NUMBER() because the performance was very poor
+                -aml.id as id,
```
And change also this.
```suggestion
                2 * aml.id as id,
```

ThiagoMForgeFlow commented:
@geraldo29 Could you attend the last reviews, please?

etobella commented:
Superseeded by #1399
